### PR TITLE
Add step to restart crio after setting up CNI

### PR DESCRIPTION
**Pull request by umohnani8**
Pod ID creation was failing with error message "cni config uninitialized." Restarting crio after the CNI setup helped fixed this.

### CODE REVIEW

**Comment by mrunalp** [comment]
@umohnani8 Are you using a packaged version of cri-o or installing from source?

**Comment by umohnani8** [comment]
@mrunalp I was installing it from source.

**Comment by umohnani8** [comment]
@mrunalp I just signed the CLA, is there a way for the pull request to check for it again?

**Comment by mrunalp** [comment]
Can we move the CNI installation section above the CRI-O installation?

**Comment by mrunalp** [comment]
I think CI should pick it up but it is still failing :/

**Comment by umohnani8** [comment]
Yeah, we could change the order also.

**Comment by mrunalp** [comment]
Verify CLA.

**Comment by mrunalp** [comment]
@umohnani8 Can you rebase and squash your commits? Also, you need to sign your commit.

File: tutorial.md [hunk 1]
```diff
@@ -60,6 +60,92 @@ commit: c91b5bea4830a57eac7882d7455d59518cdf70ec
...
 ```
 
+### cni
+
+This tutorial will use the latest version of `cni` from the master branch and build it from source.
+
+Download the `cni` source tree:
+
+```
+go get -d github.com/containernetworking/cni
```
Comment (sameo) [inline tutorial.md]: This has recently changed, and the CNI plugins are now under the plugins directory.
So this should be:
`go get -d github.com/containernetworking/plugins`

**Comment by runcom** [comment]
@sameo why do we need to restart cri-o after setting up CNI? isn't there a routine which periodically checks for network? (that's how network is in kube right?)

**Comment by sameo** [comment]
> @sameo why do we need to restart cri-o after setting up CNI? isn't there a routine which periodically checks for network? (that's how network is in kube right?)

@runcom You're right. CRI-O can start without any CNI config and once a new file under `/etc/cni/net.d` is created, it will detect it and create its default and loopback interfaces.

**Comment by rhatdan** [comment]
This has not been what we have been seeing on Fedora 25.

**Comment by runcom** [comment]
@rhatdan what do you mean? This PR doesn't look  good to us because kubelet restarts containers and restart cri-o isn't needed

**Comment by rhatdan** [comment]
When we have updated the CNI based on the CRI-O Readme. the cri-o is not getting the changes until we restart it.

**Comment by umohnani8** [comment]
I am running Fedora 25 and I always need to restart CRI-O -  this happened yesterday as well. The same thing happens if I want to create a new pod after stopping an old one. I tried it today again and ran into the same issue.

**Comment by runcom** [comment]
> I am running Fedora 25 and I always need to restart CRI-O - this happened yesterday as well. The same thing happens if I want to create a new pod after stopping an old one. I tried it today again and ran into the same issue.

I'm sorry, are you running cri-o w/o kube?

**Comment by umohnani8** [comment]
@runcom I was just following the tutorial step by step and ran into this issue at the step to create a pod. I do have kube installed though.

**Comment by sameo** [comment]
I can reproduce that, let me fix it.

**Comment by umohnani8** [comment]
@sameo When I get to the step to create a pod:
`POD_ID=$(sudo crioctl pod run --config test/testdata/sandbox_config.json)
`
This is the output I get:
`2017/06/16 12:58:10 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial unix /var/run/crio.sock: connect: no such file or directory"; Reconnecting to {"/var/run/crio.sock" <nil>}
Creating the pod sandbox failed: rpc error: code = 14 desc = grpc: the connection is unavailable
`
After I restart crio, the pod is successfully created. But when I check its status, this is the output I get:
`Name: podsandbox1
UID: redhat-test-crio
Namespace: redhat.test.crio
Attempt: 1
Status: SANDBOX_NOTREADY
Created: 2017-06-16 12:58:26.263616522 -0400 EDT`
It says sandbox is not ready, while the tutorial states it should be SANDBOX_READY.

When I stop the pod, it successfully stops. But if I try to create a new pod it will say the device or resource is busy. If I try to do the same thing again, this is the output I get:
`Creating the pod sandbox failed: rpc error: code = 2 desc = failed to delete container redhat.test.crio-podsandbox1-1-infra in pod sandbox 0d7c7b29f45c84513f98a6e46e1fd551a613592ea21b3e4087060c724ab8b42c: `/usr/bin/runc delete --force redhat.test.crio-podsandbox1-1-infra` failed: container "redhat.test.crio-podsandbox1-1-infra" does not exist
  (exit status 1)
`
I just followed the tutorial step by step, and have done it multiple times with the same result. I have opened an issue for this #586.

**Comment by runcom** [comment]
The first error you're getting is because cri-o isn't even started. I guess it's obvious we need to start it and maybe tutorial lacks such info... 

The second error is very likely selinux, did you disable selinux? 

3rd error makes me think which cri-o version youre using

**Comment by umohnani8** [comment]
@runcom The tutorial has a step to start crio, which I do. I do not get any error message so I assume it gets started?
I checked the status of SELinux and it is enabled. 
For the version, I just follow the tutorial which has a go get command for the github repo.

**Comment by runcom** [comment]
Mmm so yeah, you need to disable selinux for now, we're working to have cri-o fully working with selinux though!

**Comment by umohnani8** [comment]
Worked with @runcom to fix the issue. 
Needed to disable SELinux and get a specific version of runc. After that restarting crio or cni was not required and I was able to finish the tutorial without any problems!

**Pull request closed without merging**